dbt/include/global_project/macros/materializations/common/merge.sql

```diff
@@ -0,0 +1,30 @@
+
+
+{% macro get_merge_sql(target, source, unique_key, dest_columns) -%}
+  {{ adapter_macro('get_merge_sql', target, source, unique_key, dest_columns) }}
+{%- endmacro %}
+
+{% macro default__get_merge_sql(target, source, unique_key, dest_columns) -%}
```

Comment:
am i correct in assuming that this works for snowflake and bigquery, but not postgres and redshift? can you add adapter macros for postgres and redshift to raise an exception if this is used?

Reply:
@cmcarthur long-term, my plan is to implement a version of `merge` for Redshift and Postgres. 

The `merge` macro will serve as an abstraction that should work across all adapters, while it might compile to something like an `insert` and `update` statement on Redshift and Postgres. I really like the idea of making the core materialization logic identical across all adapters, and just calling out to an adapter-specific version of the `merge` macro.

I can definitely add an exception for pg/redshift for now though